kagenti/auth/client-registration/client_registration.py

```diff
@@ -1,72 +1,93 @@
+import json
 import os
+import jwt
 from keycloak import KeycloakAdmin, KeycloakPostError
 
+# Read SVID JWT from file to get client ID
+jwt_file_path = "/opt/jwt_svid.token"
+try:
+    with open(jwt_file_path, "r") as file:
+        content = file.read()
+
+except FileNotFoundError:
+    print(f"Error: The file {jwt_file_path} was not found.")
+except Exception as e:
+    print(f"An error occurred: {e}")
+    
+if content is None or content.strip() == "":
+    raise Exception(f'No content read from SVID JWT.')
+
+decoded = jwt.decode(content, options={"verify_signature": False})
+if 'sub' not in decoded:
+    raise Exception('SVID JWT does not contain a "sub" claim.')
+client_id = decoded['sub']
+
+
+
 KEYCLOAK_URL = os.environ.get('KEYCLOAK_URL')
+if KEYCLOAK_URL is None:
+    print(f'Expected environment variable "KEYCLOAK_URL". Skipping client registration of {client_id}.')
+    exit()
+
 KEYCLOAK_REALM = os.environ.get('KEYCLOAK_REALM')
+if KEYCLOAK_REALM is None:
+    raise Exception('Expected environment variable "KEYCLOAK_REALM"')
+
 KEYCLOAK_ADMIN_USERNAME = os.environ.get('KEYCLOAK_ADMIN_USERNAME')
+if KEYCLOAK_ADMIN_USERNAME is None:
+    raise Exception('Expected environment variable "KEYCLOAK_ADMIN_USERNAME"')
+
 KEYCLOAK_ADMIN_PASSWORD = os.environ.get('KEYCLOAK_ADMIN_PASSWORD')
+if KEYCLOAK_ADMIN_PASSWORD is None:
+    raise Exception('Expected environment variable "KEYCLOAK_ADMIN_PASSWORD"')
+
 CLIENT_NAME = os.environ.get('CLIENT_NAME')
-CLIENT_ID = os.environ.get('CLIENT_ID')
-NAMESPACE = os.environ.get('NAMESPACE')
+if CLIENT_NAME is None:
+    raise Exception('Expected environment variable "CLIENT_NAME"')
 
+
+
+# TODO: refactor this function so kagenti-client-registration image can use it
 def register_client(
-    keycloak_url: str,
-    keycloak_realm: str,
-    keycloak_admin_username: str,
-    keycloak_admin_password: str,
-    client_name: str,
+    keycloak_admin: KeycloakAdmin,
     client_id: str,
-    namespace: str
+    client_payload
 ):
-    # clientId = f'{namespace}/{client_name}'
-
-    if keycloak_url is None:
-        print(f'Expected environment variable "KEYCLOAK_URL". Skipping client registration of {client_id}.')
-        return
-    if keycloak_realm is None:
-        raise Exception('Expected environment variable "KEYCLOAK_REALM"')
-    if keycloak_admin_username is None:
-        raise Exception('Expected environment variable "KEYCLOAK_ADMIN_USERNAME"')
-    if keycloak_admin_password is None:
-        raise Exception('Expected environment variable "KEYCLOAK_ADMIN_PASSWORD"')
-    if client_name is None:
-        raise Exception('Expected environment variable "CLIENT_NAME"')
-    if client_id is None:
-        raise Exception('Expected environment variable "CLIENT_ID"')
-    if namespace is None:
-        raise Exception('Expected environment variable "NAMESPACE"')
```

Comment:
I did some refactoring to match the changes I made here previously: https://github.com/kagenti/kagenti/blob/693bd36b3c906ab3ebee93d2696f75f8712c8c2f/kagenti/auth/ui-oauth-secret/auth_secret.py#L45

Eventually, I want to refactor all these components so they utilize common libraries